src/dvc_objects/db.py

```diff
@@ -259,14 +259,11 @@ def _oids_with_limit(
         prefixes: Optional[Iterable[str]] = None,
         jobs: Optional[int] = None,
     ) -> Iterator[str]:
-        count = 0
-        for oid in self._list_oids(prefixes=prefixes, jobs=jobs):
+        for i, oid in enumerate(self._list_oids(prefixes=prefixes, jobs=jobs), start=1):
             yield oid
-            count += 1
-            if count > limit:
+            if i > limit:
```

Comment:
This probably should be
```suggestion
            if i >= limit:
```

`_oids_with_limit` has off-by-one, as it returns `limit + 1` oids. I am hesitant to change this behavior, so leaving it as is.